GitUI/UserControls/RevisionGridClasses/FormQuickGitRefSelector.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using GitUIPluginInterfaces;
+using ResourceManager;
+
+namespace GitUI.UserControls.RevisionGridClasses
+{
+    public partial class FormQuickGitRefSelector : GitModuleForm
+    {
+        private readonly TranslationString _actionRename = new TranslationString("Rename");
+        private readonly TranslationString _actionDelete = new TranslationString("Delete");
+        private readonly TranslationString _tag = new TranslationString("tag");
+
+
+        public FormQuickGitRefSelector()
+        {
+            InitializeComponent();
+            Translate();
+        }
+
+
+        /// <summary>
+        /// Gets the ref selected by the user.
+        /// </summary>
+        public IGitRef SelectedRef => (lbxRefs.SelectedItem as DisplyGitRef)?.Item;
+
+
+        public void Init(Action action, IGitRef[] refs)
+        {
+            switch (action)
+            {
+                case Action.Delete:
+                    {
+                        btnAction.Text = _actionDelete.Text;
+                    }
+                    break;
+                case Action.Rename:
+                    {
+                        btnAction.Text = _actionRename.Text;
+                    }
+                    break;
+            }
+
+            lbxRefs.Items.Clear();
+            if (refs == null || refs.Length < 1)
+            {
+                DialogResult = DialogResult.Cancel;
+                Close();
+                return;
+            }
+
+            // constrain the listbox min width, otherwise the button may not have enough space
+            var longestItemWidth = 150f;
+            try
+            {
+                lbxRefs.BeginUpdate();
+
+                using (Graphics graphics = lbxRefs.CreateGraphics())
+                {
+                    foreach (var gitRef in refs.OrderBy(r => r.IsTag).ThenBy(r => r.Name))
+                    {
+                        var suffix = gitRef.IsTag ? $" ({_tag.Text})" : string.Empty;
+                        var item = new DisplyGitRef
+                        {
+                            Label = $"{gitRef.Name}{suffix}",
+                            Item = gitRef
+                        };
+                        lbxRefs.Items.Add(item);
+
+                        // assume that the branch names or tags are never longer than 100 symbols long
+                        // if they are (sanity!) - don't resize past 100 symbols
+                        var label = item.Label.Length > 100 ? item.Label.Substring(0, 100) : item.Label;
```

Comment:
Again, giving an action a name, gives a clue what it should do. It reads better and quicker.
It can be reused - have you searched the whole code for multiple uses? What the chances are that other developer will search for such a pice of code? I think it is more likely that he/she will look for an extension method.

>OT: Arguably this isn't a one-liner, however something like return !string.IsNullOrEmpty(x);

Do we need to be so strict?

> if you wish

It is up to you. I just want to convince you not to force.